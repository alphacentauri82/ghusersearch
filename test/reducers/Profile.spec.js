import profileReducer from 'reducers/Profile';

describe('Reducer: profile', () => {

  describe('when passed no state', () => {
    it('should return the initial state', () => {
      expect(
        profileReducer(undefined, {})
      ).toMatchSnapshot();
    });
  });

  describe('when a profile request is pending', () => {
    it('should set isPending to true', () => {
      const beforeState = {
        isPending: false,
      };
      const afterState = profileReducer(beforeState, {type: 'PROFILE_REQUEST'});

      expect(
        afterState
      ).toMatchSnapshot();
    });
  });

  describe('when a profile request is successful', () => {
    it('should update the state with the userProfile', () => {
      const beforeState = {
        isPending: true,
        userProfile: {},
      };
      const action = {
        payload: {
          login: 'simonsmith',
          name: 'Simon Smith',
        },
        type: 'PROFILE_SUCCESS',
      };
      const afterState = profileReducer(beforeState, action);

      expect(
        afterState
      ).toMatchSnapshot();
      expect(afterState).not.toBe(beforeState);
    });
  });

  describe('when a search request fails', () => {
    it('should update the state with the error message', () => {
      const beforeState = {
        error: null,
        isPending: true,
      };
      const err = new Error('the server died');
      err.response = {foo: 'bar'};
      const action = {
        error: true,
        payload: err,
        type: 'PROFILE_FAILURE',
      };
      const afterState = profileReducer(beforeState, action);

      expect(
        afterState
      ).toMatchSnapshot();
      expect(afterState).not.toBe(beforeState);
    });
  });

});
