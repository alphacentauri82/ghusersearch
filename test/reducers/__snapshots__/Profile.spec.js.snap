// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer: profile when a profile request is pending should set isPending to true 1`] = `
Object {
  "isPending": true,
}
`;

exports[`Reducer: profile when a profile request is successful should update the state with the userProfile 1`] = `
Object {
  "isPending": false,
  "userProfile": Object {
    "login": "simonsmith",
    "name": "Simon Smith",
  },
}
`;

exports[`Reducer: profile when a search request fails should update the state with the error message 1`] = `
Object {
  "error": [Error: the server died],
  "isPending": false,
}
`;

exports[`Reducer: profile when passed no state should return the initial state 1`] = `
Object {
  "error": null,
  "isPending": true,
  "userProfile": Object {},
}
`;
